scenarioo-client/app/components/detailarea/detailarea.component.ts

```diff
@@ -50,11 +56,22 @@ export class DetailareaComponent {
     @Input()
     labelConfigurations: ILabelConfiguration;
 
-    @Output('valueChange')
+    @Output('togglePannelCollapsedValue')
     panelCollapsed: EventEmitter<boolean> = new EventEmitter<boolean>();
 
-    valueChange() {
+    isPanelCollapsed: boolean = true;
+
+    constructor(private localStorageService: LocalStorageService) {
+
+    }
+
+    ngOnInit(): void {
+        this.isPanelCollapsed = this.localStorageService.getBoolean(COLAPSED_STATE_KEY_PREFIX + this.key, false);
+    }
+
+    togglePannelCollapsedValue() {
         this.isPanelCollapsed = this.isPanelCollapsed === false;
+        this.localStorageService.setBoolean(COLAPSED_STATE_KEY_PREFIX + this.key, this.isPanelCollapsed);
```

Comment:
I would extract "COLAPSED_STATE_KEY_PREFIX + this.key" into a method getLocalStorageKey() to prevent that the key concatenation is changed in only one of the two places (L69 & L74).